### DISCUSSION TITLE
How to simulate or test autoReplaceStringTemplate replacement result locally

### DISCUSSION THREAD

**Discussion by bdovaz** [renovatebot/renovate#29988] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure Pipelines

### Please tell us more about your question or problem

I am trying to use this option: https://docs.renovatebot.com/configuration-options/#autoreplacestringtemplate

But my doubt is, how can I verify the result of the replace?

Because running it locally: https://docs.renovatebot.com/modules/platform/local/

I can verify that my custom regex manager finds results, if I pretend to be in an old version of a dependency I see that there is an update but I can't see in [debug](https://docs.renovatebot.com/troubleshooting/#log-debug-levels) the result of the replace.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You cannot see the replaced string in the debug logs as it is not logged. You need to create a test repo and verify the replacements work as expected.

That said the debug logs where updates are shown, have the `depName`, `replaceString`, `newValue` and the `autoReplaceStringTemplate` , etc.. which should give you an idea of the replacement.
